internals/approval_defs.py

```diff
@@ -153,7 +155,61 @@ def is_resolved(approval_values, field_id):
 
   # Any NOT_APPROVED value means that the review is no longer pending.
   for av in approval_values:
-    if av.state == review_models.Approval.NOT_APPROVED:
+    if av.state == Approval.NOT_APPROVED:
       return True
 
   return False
+
+def set_vote(
+    feature_id: int, gate_id: int, new_state: int, set_by_email: str) -> None:
+  """Add or update an approval value."""
+  if not Vote.is_valid_state(new_state):
+    raise ValueError('Invalid approval state')
+
+  now = datetime.datetime.now()
+  existing_list: list[Vote] = Vote.get_votes(
+      feature_id=feature_id, gate_id=gate_id, set_by=set_by_email)
+  if existing_list:
+    existing = existing_list[0]
+    existing.set_on = now
+    existing.state = new_state
+    existing.put()
+    return
```

Comment:
I think you need an `else` here instead of `return`.   With `return` the code might not reach the next section of code that updates the gate.

Reply:
This is true - I made this change in a previous PR, but I think it got lost in translation up to this point. Fixed!